frontend/public/components/modals/cluster-update-modal.tsx

```diff
@@ -36,14 +51,46 @@ const ClusterUpdateModal = withHandlePromise((props: ClusterUpdateModalProps) =>
       ? currentMinorVersionPatchUpdate?.version
       : availableSortedUpdates[0]?.version) || '',
   );
+  const [machineConfigPools, machineConfigPoolsLoaded] = useK8sWatchResource<
+    MachineConfigPoolKind[]
+  >({
+    isList: true,
+    kind: referenceForModel(MachineConfigPoolModel),
+  });
   const [error, setError] = React.useState(errorMessage);
   const [isOpen, setIsOpen] = React.useState(false);
-  const onToggle = () => setIsOpen(!isOpen);
-  const onSelect = (event, selection) => {
+  const [machineConfigPoolsToPause, setMachineConfigPoolsToPause] = React.useState<string[]>([]);
+  const [upgradeType, setUpgradeType] = React.useState<upgradeTypes>(upgradeTypes.Full);
+  React.useEffect(() => {
+    const initialMCPPausedValues = [];
+    machineConfigPools.map((mcp) => {
+      !isMCPMaster(mcp) && isMCPPaused(mcp)
+        ? initialMCPPausedValues.push(mcp.metadata.name)
+        : _.pull(mcp.metadata.name);
```

Comment:
You shouldn't really ever do a `.push` on an array in the middle of a `.map`. What you want is `.forEach` I think in this case... Personally I favour a `.reduce` as you're building an array that can dynamically grow and shrink as you iterate over the data. 

If you're just looking at only keeping values that are not master & are paused... I think you just want a `.filter`, no?

Reply:
Thanks.  Will change that `map` to a `forEach` as in this case I'm creating a new array of strings (going to add a type on line 65 also) in order to set which MCPs are paused when the modal is launched (if any).  In theory the `pull` will never occur because the array is initialized to empty each to the effect runs, but figured it was a good measure.  Is there another approach I should consider?

Reply:
If you're looking to take a list of resources, turn it into a list of their `.metadata.name`s and omit anything that doesn't fit a given state (ie `!isMCPMaster(mcp) && isMCPPaused(mcp)`) there is a simple approach: `[].filter().map()`:

```
const initialMCPPausedValues = machineConfigPools
  .filter((mcp) => !isMCPMaster(mcp) && isMCPPaused(mcp))
  .map((mcp) => mcp.metadata.name)
```
Which is easy to read as "taking machineConfigPool resources, filter ones we don't care about, just get the name of the resource". Now you have a simple list of strings ready to go.

Reply:
The reason I didn't suggest this earlier was I was not sure if your `pull` was intending to try and de-duplicate values that were previously added in a previous loop of `.map`.

Reply:
Updated to your suggestion and limited to effect to run once.  It occurred to me we don't want to change the checkboxes once they are rendered because that could inadvertently override user input.